### ISSUE TITLE
MIGRAPHX_TRACE_EVAL segmentation fault

### ISSUE THREAD

**Issue by mvermeulen** [ROCmSoftwarePlatform/AMDMIGraphX#960]
The following seems to work as expected
```
env MIGRAPHX_TRACE_EVAL=1 ./driver run resnet50i1.onnx 
```
while the following results in segmentation fault
```
env MIGRAPHX_TRACE_EVAL=2 ./driver run resnet50i1.onnx
```
Last lines of the output are as follows
```
Allocating params ... 
Run instruction: main:@0 = check_context::migraphx::version_1::gpu::context -> float_type, {}, {}
Time: 0.003787ms, 0.004979ms
Output: 
Run instruction: main:@1 = hip::hip_allocate_memory[shape=float_type, {2408448}, {1},id=main:scratch] -> float_type, {2408448}, {1}
Time: 0.004388ms, 0.00514ms
Segmentation fault
```

**Comment by scxiao** [comment]
The segmentation fault is caused by printing a result in the gpu memory.

**Comment by scxiao** [comment]
@pfultz2 To resolve the above problem, I think we can either adding a from_gpu() method in the context or store a target obj in the prog_impl struct. Not sure whether you have other better way....

**Comment by scxiao** [comment]
Resolved in PR#974, so close this issue.
